src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -161,7 +161,12 @@ public static void DeleteFile(string fullPath)
                 switch (errorInfo.Error)
                 {
                     case Interop.Error.ENOENT:
-                        // ENOENT means it already doesn't exist; nop
+                        // ENOENT means it already doesn't exist; nop         
+                        var directoryName = Path.GetDirectoryName(fullPath);
+                        if(!string.IsNullOrEmpty(directoryName) && !Directory.Exists(directoryName))
```

Comment:
Surely it will never return null to you . And hopefully Exists quickly returns for empty string

Reply:
> Surely it will never return null to you . And hopefully Exists quickly returns for empty string 

I don't understand your proposal, do you prefere remove `IsNullOrEmpty` and leave only `Exists`?

Reply:
But we need to check for empty out of `Exit` because return false and throw DNFE, right?

Reply:
You're right, I guess  you can remove the null check though